WcaOnRails/app/webpacker/components/EditSchedule/EditActivities/EditActivityModal.js

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect } from 'react';
+import {
+  Button,
+  Container,
+  Form,
+  Modal,
+} from 'semantic-ui-react';
+import { DateTime } from 'luxon';
+import useInputState from '../../../lib/hooks/useInputState';
+
+const commonActivityCodes = {
+  'other-registration': 'On-site registration',
+  'other-checkin': 'Check-in',
+  'other-tutorial': 'Tutorial for new competitors',
+  'other-multi': 'Cube submission for 3x3x3 Multi-Blind',
+  'other-breakfast': 'Breakfast',
+  'other-lunch': 'Lunch',
+  'other-dinner': 'Dinner',
+  'other-awards': 'Awards',
+  'other-misc': 'Other',
+};
+
+const otherActivityCodeOptions = Object.keys(commonActivityCodes).map((activityCode) => ({
+  key: activityCode,
+  text: commonActivityCodes[activityCode],
+  value: activityCode,
+}));
+
+function EditActivityModal({
+  showModal,
+  activity,
+  startLuxon,
+  endLuxon,
+  onModalClose,
+  dateLocale,
+}) {
+  const [activityCode, setActivityCode] = useInputState();
+  const [activityName, setActivityName] = useInputState();
+
+  const setActivityCodeInternal = (evt, data) => {
+    const { value: newActivityCode } = data;
+
+    // only if there is no name yet: assign a default name based on the activity code
+    if (!activityName && newActivityCode) {
+      setActivityName(commonActivityCodes[newActivityCode]);
+    }
+
+    setActivityCode(evt, data);
+  };
+
+  // We have to assign state in this awkward way because of an opinionated conflict
+  //   between FullCalendar and SemanticUI. See comment at the beginning of index.js for details.
+  useEffect(() => {
+    setActivityCode(activity?.activityCode);
+    setActivityName(activity?.name);
+  }, [activity, setActivityCode, setActivityName]);
```

Comment:
You should be able to omit `setActivityCode` and `setActivityName` from the dependency array; React knows they don't change.

Reply:
Surprisingly enough, this is ESLint which insists that I have to put them in there:

```
ESLint: React Hook useEffect has missing dependencies: 'setActivityCode' and 'setActivityName'. Either include them or remove the dependency array.(react-hooks/exhaustive-deps)
```